src/Microsoft.DocAsCode.Common/FileAbstractLayer/PathMapping.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Common
+{
+    using System;
+    using System.Collections.Immutable;
+
+    public struct PathMapping : IEquatable<PathMapping>
+    {
+        public PathMapping(RelativePath logicalPath, string physicalPath)
+        {
+            if (logicalPath == null)
+            {
+                throw new ArgumentNullException(nameof(logicalPath));
+            }
+            if (physicalPath == null)
+            {
+                throw new ArgumentNullException(nameof(physicalPath));
+            }
+            LogicalPath = logicalPath.GetPathFromWorkingFolder();
+            PhysicalPath = physicalPath;
+            AllowMoveOut = false;
+            Properties = ImmutableDictionary<string, string>.Empty;
+        }
+
+        public RelativePath LogicalPath { get; }
+
+        public string PhysicalPath { get; }
+
+        public bool IsFolder => LogicalPath.FileName == string.Empty;
+
+        public bool AllowMoveOut { get; set; }
+
+        public ImmutableDictionary<string, string> Properties { get; set; }
+
+        #region Equals & operators
+
+        public override int GetHashCode()
```

Comment:
HashCode equals not means Equals.

---
In reply to: [89576952](https://github.com/dotnet/docfx/pull/926#discussion_r89576952) [](ancestors = 89576952)